pkg/codegen/docs/README.md

```diff
@@ -0,0 +1,40 @@
+# Docs generator
+
+This generator generates resource-level docs by utilizing the Pulumi schema generated by [tfbridge](https://github.com/pulumi/pulumi-terraform-bridge/).
+
+## Crash course on templates
+
+The templates use Go's built-in `html/template` package to process templates with data. The driver for this doc generator (in tfbridge) then persists each file from memory onto the disk as `.md` files in the output directory specified by `tfbridge`.
```

Comment:
> Then, in the docs repo, create a new go docgen tool that has pulumi/pulumi as a dependency, that can take the Pulumi schema as input (standard input?) and an argument specifying the output directory, and "drive" the doc gen from the input schema?
>
> The scripts in the docs gen for kicking this all off for each repo, would be coded to call tfbridge to get the schema for TF-based providers, and whatever command line tool is required to get the schema for Kubernetes.

I like this idea, and in fact, in a branch for the `docs` repo I have something like this. See this [Makefile](https://github.com/pulumi/docs/blob/praneetloke/docs/prototype/Makefile#L40) target and this [script](https://github.com/pulumi/docs/blob/praneetloke/docs/prototype/scripts/gen_resource_docs.sh), that it calls. I do think that the `docs` repo should be the only "driver" of the docs generator in this repo. I think it makes it easier to understand the direct dependencies too, rather than transitively calling this tool.

> Or instead of a new go-based docgen tool in the docs repo, we could consider just adding another hidden command to the Pulumi CLI itself, similar to the hidden gen-markdown command we have to generate docs for the CLI commands. I could see a hidden gen-docs command, that takes the Pulumi schema from standard input and outputs to a specified directory.

Yeah this is fine too I guess. Not sure if there is a strong reason to do this over the above approach, though? And if not, I'd prefer to do the above. What do you think?

Reply:
The above approach sounds good to me.